week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/models/castle.ts

```diff
@@ -0,0 +1,18 @@
+import { Dungeon } from "./dungeon";
+import { Person } from "./person";
+
+export class Castle {
+  constructor(private dungeon: Dungeon) {
+  }
+
+  gotoTheDungeon(daredevil: Person) {
+    // #RQ1: peasant cannot be at the Castle !
+    if (daredevil.title === "peasant") {
+      console.error('\x1b[31m%s\x1b[0m', "Peasant cannot be at the Castle!");
```

Comment:
Generalnie @teziovsky zadanie jest do poprawki o "obsługę błędów".

Chodzi o to że logika działa dobrze jednak:
> 2. Odpowiednio obsługuj błędy - od Ciebie zależy jaką będą miały treść

---

> Z poziomu organizowanej misji (1-3) - powinniśmy umieć odpowiedzieć na pytanie - dlaczego nie udało się uwolnić księżniczki. Co poszło nie tak i gdzie się to wydarzyło.

---

Oznacza to, że z poziomu misji musisz wiedzieć co się wydarzyło, a tutaj jesteśmy np. w `Castle`.
Czyli ten `console.error` jest ok, ale powinien być na poziomie `main.ts` w momencie w którym załapiesz wyjątek.

Reply:
Okej zmieniłem obsługę błędów na `throw new Error` przez co z poziomu `main.ts`, a w zasadzie `mission.ts`, bo wyodrębniłem logikę poszczególnych misji do osobnego pliku, jest teraz wyświetlanie błędów za pomocą `console.error` 😎